flashlight/fl/contrib/modules/VisionTransformer.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "flashlight/fl/contrib/modules/VisionTransformer.h"
+
+#include <cmath>
+
+#include "flashlight/fl/autograd/Functions.h"
+#include "flashlight/fl/nn/Init.h"
+#include "flashlight/fl/nn/Utils.h"
+
+namespace {
+
+const float pi = std::acos(-1);
+const float geluConst1 = std::sqrt(2 / pi);
+const float geluConst2 = 0.044715;
+
+} // namespace
+
+namespace fl {
+
+VisionTransformer::VisionTransformer(
+    int32_t modelDim,
+    int32_t headDim,
+    int32_t mlpDim,
+    int32_t nHeads,
+    float pDropout,
+    float pLayerdrop)
+    : nHeads_(nHeads),
+      pDropout_(pDropout),
+      pLayerdrop_(pLayerdrop),
+      w1_(std::make_shared<Linear>(initLinear(modelDim, mlpDim))),
+      w2_(std::make_shared<Linear>(initLinear(mlpDim, modelDim))),
+      // wq_(std::make_shared<Linear>(
+      //     initLinear(modelDim, headDim * nHeads),
+      //     fl::constant(0., headDim * nHeads, 1))),
+      // wk_(std::make_shared<Linear>(
+      //     initLinear(modelDim, headDim * nHeads),
+      //     fl::constant(0., headDim * nHeads, 1))),
+      // wv_(std::make_shared<Linear>(
+      //     initLinear(modelDim, headDim * nHeads),
+      //     fl::constant(0., headDim * nHeads, 1))),
+      wq_(std::make_shared<Linear>(initLinear(modelDim, headDim * nHeads))),
```

Comment:
all xavier https://github.com/google-research/vision_transformer/blob/00883dd691c63a6830751563748663526e811cee/vit_jax/models.py#L129, bias is zero